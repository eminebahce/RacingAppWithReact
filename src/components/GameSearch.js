import React, {Component} from 'react';
import axios from 'axios';
import {Button, Card, CardBody, CardImg, CardSubtitle, CardText, CardTitle, Col, Row} from 'reactstrap';

const PRODUCTS_URL = `https://www.atg.se/services/racinginfo/v1/api/products/`;

class GameSearch extends Component {
    constructor(props) {
        super(props);

        this.state = ({
            upcoming: [],
            results: []
        });
    }

    gameSearchByType(gameType) {
        const url = `${PRODUCTS_URL}${gameType}`;
        axios.get(url)
            .then(res => {
                this.setState({
                    upcoming: res.data.upcoming,
                    results: res.data.results
                });
            });
    }

    render() {
        return (
            <div className="formatCardClass1">
                <Row className="formatCardClass1">
                    <Col sm="3">
                        <Card>
                            <CardImg top width="25%"
                                     src="http://www.swedishhorseracing.com/assets/images/racethumbs/v75.png"
                                     alt="V75 image"/>
                            <CardBody>
                                <CardTitle>Game V75</CardTitle>
                                <CardSubtitle></CardSubtitle>
                                <CardText>Click button to see the schedule table for game type V75</CardText>
                                <Button onClick={event => this.gameSearchByType(event.target.textContent)}
                                        className="btn btn-success">
                                    V75
                                </Button>
                            </CardBody>
                        </Card>
                    </Col>
                    <Col sm="3">
                        <Card>
                            <CardImg top width="25%"
                                     src="http://www.swedishhorseracing.com/assets/images/racethumbs/v65.png"
                                     alt="V65 image"/>
                            <CardBody>
                                <CardTitle>Game V65</CardTitle>
                                <CardSubtitle></CardSubtitle>
                                <CardText>Click button to see the schedule table for game type V65</CardText>
                                <Button onClick={event => this.gameSearchByType(event.target.textContent)}
                                        className="btn btn-success">
                                    V65
                                </Button>
                            </CardBody>
                        </Card>
                    </Col>
                    <Col sm="3">
                        <Card>
                            <CardImg top width="25%"
                                     src="http://www.swedishhorseracing.com/assets/images/racethumbs/v64.png"
                                     alt="V64 image"/>
                            <CardBody>
                                <CardTitle>Game V64</CardTitle>
                                <CardSubtitle></CardSubtitle>
                                <CardText>Click button to see the schedule table for game type V64</CardText>
                                <Button onClick={event => this.gameSearchByType(event.target.textContent)}
                                        className="btn btn-success">
                                    V64
                                </Button>
                            </CardBody>
                        </Card>
                    </Col>
                    <Col sm="3">
                        <Card>
                            <CardImg top width="25%"
                                     src="http://www.swedishhorseracing.com/assets/images/racethumbs/v4.png"
                                     alt="V4 image"/>
                            <CardBody>
                                <CardTitle>Game V4</CardTitle>
                                <CardSubtitle></CardSubtitle>
                                <CardText>Click button to see the schedule table for game type V4</CardText>
                                <Button onClick={event => this.gameSearchByType(event.target.textContent)}
                                        className="btn btn-success">
                                    V4
                                </Button>
                            </CardBody>
                        </Card>
                    </Col>
                </Row>
                <Row className="formatCardClass1">
                    <Col sm="12">

                    </Col>
                </Row>
            </div>
        );
    }

}

export default GameSearch;
