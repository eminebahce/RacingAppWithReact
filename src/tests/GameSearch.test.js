import React from 'react';
import Renderer from 'react-test-renderer';
import GameSearch from '../components/GameSearch';

test('GameSearch component renders gameSearch correctly', () => {
    const component = Renderer.create(
        <GameSearch/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
