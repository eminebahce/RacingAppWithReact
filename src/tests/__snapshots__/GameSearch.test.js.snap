// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameSearch component renders gameSearch correctly 1`] = `
<div
  className="formatCardClass1"
>
  <div
    className="formatCardClass1 row"
  >
    <div
      className="col-sm-3"
    >
      <div
        className="card"
      >
        <img
          alt="V75 image"
          className="card-img-top"
          src="http://www.swedishhorseracing.com/assets/images/racethumbs/v75.png"
          width="25%"
        />
        <div
          className="card-body"
        >
          <h5
            className="card-title"
          >
            Game V75
          </h5>
          <h6
            className="card-subtitle"
          />
          <p
            className="card-text"
          >
            Click button to see the schedule table for game type V75
          </p>
          <button
            className="btn btn-success btn btn-secondary"
            onClick={[Function]}
            type="button"
          >
            V75
          </button>
        </div>
      </div>
    </div>
    <div
      className="col-sm-3"
    >
      <div
        className="card"
      >
        <img
          alt="V65 image"
          className="card-img-top"
          src="http://www.swedishhorseracing.com/assets/images/racethumbs/v65.png"
          width="25%"
        />
        <div
          className="card-body"
        >
          <h5
            className="card-title"
          >
            Game V65
          </h5>
          <h6
            className="card-subtitle"
          />
          <p
            className="card-text"
          >
            Click button to see the schedule table for game type V65
          </p>
          <button
            className="btn btn-success btn btn-secondary"
            onClick={[Function]}
            type="button"
          >
            V65
          </button>
        </div>
      </div>
    </div>
    <div
      className="col-sm-3"
    >
      <div
        className="card"
      >
        <img
          alt="V64 image"
          className="card-img-top"
          src="http://www.swedishhorseracing.com/assets/images/racethumbs/v64.png"
          width="25%"
        />
        <div
          className="card-body"
        >
          <h5
            className="card-title"
          >
            Game V64
          </h5>
          <h6
            className="card-subtitle"
          />
          <p
            className="card-text"
          >
            Click button to see the schedule table for game type V64
          </p>
          <button
            className="btn btn-success btn btn-secondary"
            onClick={[Function]}
            type="button"
          >
            V64
          </button>
        </div>
      </div>
    </div>
    <div
      className="col-sm-3"
    >
      <div
        className="card"
      >
        <img
          alt="V4 image"
          className="card-img-top"
          src="http://www.swedishhorseracing.com/assets/images/racethumbs/v4.png"
          width="25%"
        />
        <div
          className="card-body"
        >
          <h5
            className="card-title"
          >
            Game V4
          </h5>
          <h6
            className="card-subtitle"
          />
          <p
            className="card-text"
          >
            Click button to see the schedule table for game type V4
          </p>
          <button
            className="btn btn-success btn btn-secondary"
            onClick={[Function]}
            type="button"
          >
            V4
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="formatCardClass1 row"
  >
    <div
      className="col-sm-12"
    >
      <div>
        <div>
          
        </div>
        <div>
          <div
            className="formatCardClass2"
          >
            <div
              className="row"
            >
              <div
                className="col-sm-12"
              >
                <div
                  className="text-center"
                >
                  <div
                    className="btn-group"
                  >
                    <div
                      className="btn-group"
                      role="group"
                    >
                      
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="formatCardClass3 row"
            >
              <div
                className="col-sm-12"
              >
                <div>
                  
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
